examples/inverse/plot_mne_cov_power.py

```diff
@@ -0,0 +1,112 @@
+# Author: Denis A. Engemann <denis-alexander.engemann@inria.fr>
+#         Luke Bloy <luke.bloy@gmail.com>
+#
+# License: BSD (3-clause)
+
+import os.path as op
+import numpy as np
+
+import mne
+from mne.datasets import sample
+from mne.minimum_norm import make_inverse_operator, apply_inverse_cov
+
+data_path = sample.data_path()
+subjects_dir = data_path + '/subjects'
+
+raw_fname = data_path + '/MEG/sample/sample_audvis_filt-0-40_raw.fif'
+raw = mne.io.read_raw_fif(raw_fname)  # already has an average reference
+raw.load_data()
+
+raw_empty_room_fname = op.join(
+    data_path, 'MEG', 'sample', 'ernoise_raw.fif')
+raw_empty_room = mne.io.read_raw_fif(raw_empty_room_fname)
+raw_empty_room.crop(0, 60)
+raw_empty_room.info['bads'] = ['MEG 2443']
+raw_empty_room.info['projs'] = raw.info['projs']
+
+events = mne.find_events(raw, stim_channel='STI 014')
+
+# event trigger and conditions
+event_id = dict(aud_l=1, aud_r=2, vis_l=3, vis_r=4)
+tmin, tmax = -0.2, 0.5
+raw.info['bads'] = ['MEG 2443', 'EEG 053']
+baseline = (None, 0)  # means from the first instant to t = 0
+reject = dict(grad=4000e-13, mag=4e-12, eog=150e-6)
+
+epochs_noise = mne.Epochs(raw, events, event_id, tmin, tmax=0., proj=True,
+                          picks=('meg', 'eog'), baseline=None, reject=reject)
+
+noise_cov = mne.compute_raw_covariance(
+    raw_empty_room, method=['empirical', 'shrunk'])
+
+raw.filter(4, 12)
+epochs = mne.Epochs(raw.copy().filter(4, 12), events, event_id, tmin, tmax,
+                    proj=True, picks=('meg', 'eog'), baseline=None,
+                    reject=reject, preload=True)
+
+data_cov = mne.compute_covariance(
+    epochs, tmin=0., tmax=0.2, method=['shrunk', 'empirical'], rank=None,
+    verbose=True)
+
+base_cov = mne.compute_covariance(
+    epochs, tmin=-0.2, tmax=0, method=['shrunk', 'empirical'], rank=None,
+    verbose=True)
+
+fig_noise_cov = mne.viz.plot_cov(noise_cov, raw.info, show_svd=False)
+fig_cov = mne.viz.plot_cov(data_cov, raw.info, show_svd=False)
+
+evoked = epochs.average().pick('meg')
+evoked.drop_channels(evoked.info['bads'])
+evoked.plot(time_unit='s')
+evoked.plot_topomap(times=np.linspace(0.05, 0.15, 5), ch_type='mag',
+                    time_unit='s')
+
+evoked_noise_cov = mne.EvokedArray(data=np.diag(noise_cov['data'])[:, None],
+                                   info=evoked.info)
+
+evoked_data_cov = mne.EvokedArray(data=np.diag(data_cov['data'])[:, None],
+                                  info=evoked.info)
+
+evoked_data_cov_white = mne.whiten_evoked(evoked_data_cov, noise_cov)
+
+
+def plot_cov_diag_topomap(evoked, ch_type='grad'):
+    evoked.plot_topomap(
+        ch_type=ch_type, times=[0],
+        vmin=np.min, vmax=np.max, cmap='viridis',
+        units=dict(mag='None', grad='None'),
+        scalings=dict(mag=1, grad=1),
+        cbar_fmt=None)
+
+
+plot_cov_diag_topomap(evoked_noise_cov, 'grad')
+plot_cov_diag_topomap(evoked_data_cov, 'grad')
+plot_cov_diag_topomap(evoked_data_cov_white, 'grad')
+
```

Comment:
I think some explanatory comments on how the covariance matrices are estimated here and what the plots tell us could be really beneficial, especially for novice users.

Reply:
+1

Reply:
this example also is not structured in sections. Adding sections that explain what is done would help pedagogy.

see https://18288-1301584-gh.circle-artifacts.com/0/dev/auto_examples/inverse/plot_mne_cov_power.html#sphx-glr-auto-examples-inverse-plot-mne-cov-power-py